src/nvim/terminal.c

```diff
@@ -1151,11 +1179,8 @@ static void refresh_terminal(Terminal *term)
 static void refresh_timer_cb(TimeWatcher *watcher, void *data)
 {
   refresh_pending = false;
-  if (exiting  // Cannot redraw (requires event loop) during teardown/exit.
-      || (State & CMDPREVIEW)
-      // WM_LIST (^D) is not redrawn, unlike the normal wildmenu. So we must
-      // skip redraws to keep it visible.
-      || wild_menu_showing == WM_LIST) {
+  // do we really call time watchers when exiting?
```

Comment:
yes

Reply:
but should we? what in exiting depends on time watchers?

Reply:
could be an ordering issue, but in general it's because we "drain" the uv loops before exiting.

Reply:
That the raw uv callbacks are called are one thing, but do we need to call the deferred main loop handlers like this one?

Reply:
I guess it was to avoid a race if user (or a test) wants to display something in the terminal just before exiting.

Reply:
ok, let's keep this for now then